presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -96,8 +104,28 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
 
             ImmutableList.Builder<ConnectorSplit> splits = ImmutableList.builder();
 
+            Domain partitionIdDomain = null;
+            Domain partitionOffsetDomain = null;
```

Comment:
`partitionOffsetDomain` should have partition ID information, otherwise, the following test fails:

```java
assertQuery(
  "SELECT _partition_id, min(_partition_offset), max(_partition_offset) FROM orders WHERE (_partition_id = 1 AND _partition_offset BETWEEN 100 AND 200) OR (_partition_id = 0 AND _partition_offset BETWEEN 4000 AND 5000) GROUP BY _partition_id",
  "VALUES (0, 4000, 5000), (1, 100, 200)");
```

I think the following code is helpful:

https://github.com/reproio/presto/blob/75393474fa589a1d6aa54c1b3b8b26efe1694b4a/presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java#L128-L169

Reply:
Thanks!

I fixed a way for collecting split range with `com.facebook.presto.spi.predicate.Range`